### DISCUSSION TITLE
The API v1/images/generations do not apply the size

### DISCUSSION THREAD

**Discussion by thuongvovan** [mudler/LocalAI#2431] [Q&A]
**LocalAI version:**

localai/localai:v2.16.0-cublas-cuda12

**Environment, CPU architecture, OS, and Version:**

Linux Ubuntu-2204-jammy-amd64-base 5.15.0-107-generic #117-Ubuntu SMP Fri Apr 26 12:26:49 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

**Describe the bug**

The API v1/images/generations do not apply the size. All of the responses is 512x512 for many different input values. I have changed to another model but it is the same.

**To Reproduce**

**Expected behavior**

**Logs**

**Additional context**

**Comment by mudler** [comment]
which models have you tried? note that only SDXL models support more than 512x512.

**Reply by thuongvovan** [reply]
Thank you for considering my question.
I have tried some models like: stable-diffusion-xl-base, stable-diffusion-v1-5, dreamShaper... I have set the size lower 512x512 for example 256x256, but it still returns a 512x512 image, my expectation in this case is a 256x256 image.
If I have misunderstood something, please forgive me.
I would like to say thank you again for the wonderful product that you made and what good you have contributed for community. It is very helpful for a newbie in AI like me.

**Reply by mudler** [reply]
I'm testing things here but it looks the size is passed correctly:

```
9:56AM DBG Input: &{PredictionOptions:{Model: Language: N:0 TopP:<nil> TopK:<nil> Temperature:<nil> Maxtokens:<nil> Echo:false Batch:0 IgnoreEOS:false RepeatPenalty:0 Keep:0 FrequencyPenalty:0 PresencePenalty:0 TFZ:<nil> TypicalP:<nil> Seed:<nil> NegativePrompt: RopeFreqBase:0 RopeFreqScale:0 NegativePromptScale:0 UseFastTokenizer:false ClipSkip:0 Tokenizer:} Context:context.Background.WithCancel Cancel:0x4ce5e0 File: ResponseFormat:{Type:} Size:256x256 Prompt:A cute baby sea otter Instruction: Input:<nil> Stop:<nil> Messages:[] Functions:[] FunctionCall:<nil> Tools:[] ToolsChoice:<nil> Stream:false Mode:0 Step:0 Grammar: JSONFunctionGrammarObject:<nil> JSONFunctionGrammarObjectName:<nil> Backend: ModelBaseName:}
```

Are you sure you are passing the size correctly? can you share your settings/curl calls?

**Reply by thuongvovan** [reply]
Yes. It is almost like your default. With the call in the image below return 512x512 image in both.

![CleanShot 2024-05-29 at 15 12 55@2x](https://github.com/mudler/LocalAI/assets/54501918/707049e6-e014-4b1b-b086-7047f2ec3a7b)
